crates/router/src/macros.rs

```diff
@@ -51,3 +51,18 @@ macro_rules! async_spawn {
         tokio::spawn(async move { $t });
     };
 }
+
+#[macro_export]
+macro_rules! get_keys_of_option_nones {
```

Comment:
We can probably name this better. Maybe something like `collect_missing_value_keys`? Feel free to be creative!

Reply:
The name `get_keys_of_option_nones` is explicit of having a collection of tuples with `[(&'static str, Option<_>)]`. keeping the generic nature.  
I am happy to change the name but naming as `missing values` express the context of macro usage.

Reply:
@vspecky I am not sure at the moment if renaming is required. @SanchithHegde @Narayanbhat166  what do you guys suggest?

Reply:
Not really a hard-and-fast requirement per-se, more of an opinion. Personally I find it a bit hard to glean the purpose of the macro from the name, so suggested exploring better names. If we can't do it without being too verbose, or if others think the name's alright, then I'm okay with it!

Reply:
You can rename it, not a problem. I feel `collect_missing_value_keys` has the necessary info about what the macro does: returns the missing fields. If you can come up with a better name that indicates the purpose of the macro, you can switch to it too.

Reply:
coolio! renaming the macro.